src/Quartz/DefaultDirectoryProvider.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Quartz.Job;
+
+namespace Quartz
+{
+    internal class DefaultDirectoryProvider : IDirectoryProvider
+    {
+        public IEnumerable<string> GetDirectoriesToScan(JobDataMap mergedJobDataMap)
+        {
+            List<string> directoriesToScan = new List<string>();
+            string dirName = mergedJobDataMap.GetString(DirectoryScanJob.DirectoryName);
+            string dirNames = mergedJobDataMap.GetString(DirectoryScanJob.DirectoryNames);
+
+            if (dirName == null && dirNames == null)
```

Comment:
Right now if the user supplies neither `DIRECTORY_NAME` nor `DIRECTORY_NAMES`, [then a `JobExecutionException` is thrown](https://github.com/cknightdevelopment/quartznet/blob/directory-scanner-updates/src/Quartz/DefaultDirectoryProvider.cs#L18). If both are supplied, then currently only `DIRECTORY_NAME` is used. Perhaps, we should just use both in that case (removing duplicates)? Or would you rather throw an exception?